applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -995,18 +1004,7 @@ void CrBeamElement3D2N::CalculateLocalSystem(MatrixType &rLeftHandSideMatrix,
   // add bodyforces
   noalias(rRightHandSideVector) += this->CalculateBodyForces();
   this->mIterationCount++;

```

Comment:
this is delicate, we have to check in detail

Reply:
I understand, In fact I see many code that can be refactored between the 2D/3D version, and many things that can be maybe simplified, as using an internal counter (mIterationCount) instead of a flag